triton-docker

```diff
@@ -71,6 +59,58 @@ function install () {
     echo
 }
 
+# Emit the current Triton profile and time
+function currentProfile () {
+    if [ -n "$TRITON_PROFILE" ]
+    then
+        local profile_source='via env var; use `eval "$(triton env <profile name>)"` to change'
+    else
+        local profile_source='default; use `triton profile set <profile name>` to change'
+    fi
+
+    echo "Executing in '$(triton profile get | awk 'NR == 1 {print $2}')' (${profile_source})"
+}
+
+# Extra env vars
+function extraEnvVars () {
+
+    # If the CNS env vars are already set, don't bother continuing
+    if [ -n "$TRITON_CNS_SUFFIX_PUBLIC" ] && [ -n "$TRITON_CNS_SUFFIX_PRIVATE" ]
+    then
+        return
+    fi
+
+    # Get and the user's account information for parsing later
+    local triton_account="$(triton account get)"
+
+    # Check if CNS is enabled, require it
+    local triton_cns_enabled="$(echo "$triton_account" | awk -F": " '/cns/{print $2}')"
+    if [ ! "true" == "$triton_cns_enabled" ]; then
+        echo
+        tput rev  # reverse
+        tput bold # bold
+        echo 'Error! Triton CNS is required and not enabled.'
+        tput sgr0 # clear
+        echo
+        echo 'Please run:'
+        echo '  triton account update triton_cns_enabled=true'
+        echo
+        exit 1
+    fi
+
+    # Get the user's UUID
+    local triton_account_uuid="$(echo "$triton_account" | awk -F": " '/id:/{print $2}')"
+
+    # Get current data center name
+    # Note: this makes assumptions that work in our public cloud, but might not work elsewhere
+    # Further note: set TRITON_CNS_SUFFIX_PUBLIC and TRITON_CNS_SUFFIX_PRIVATE to work around this
+    local triton_dc="$(triton profile get | awk -F"/" '/url:/{print $3}' | awk -F'.' '{print $1}')"
+
+    # Set the CNS base for public and private names
+    export TRITON_CNS_SUFFIX_PUBLIC="${triton_account_uuid}.${triton_dc}.triton.zone"
```

Comment:
I love that we're trying to make things easier but part of why I'd suggested just the account UUID and DC be added is because the CNS suffix is going to be specific to the installation; this wouldn't work with private cloud installs.